plugins/woocommerce-admin/client/products/fields/category-field/category-field.tsx

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useMemo, useState } from '@wordpress/element';
+import { Popover } from '@wordpress/components';
+import {
+	Spinner,
+	__experimentalSelectControl as SelectControl,
+	__experimentalSelectControlItem as SelectControlItem,
+} from '@woocommerce/components';
+import { ProductCategory } from '@woocommerce/data';
+import { debounce } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './category-field.scss';
+import { CategoryFieldItem, CategoryTreeItem } from './category-field-item';
+import { useCategorySearch } from './use-category-search';
+import { CategoryFieldAddNewItem } from './category-field-add-new-item';
+import { CreateCategoryModal } from './create-category-modal';
+
+type CategoryFieldProps = {
+	label: string;
+	placeholder: string;
+	value?: Pick< ProductCategory, 'id' | 'name' >[];
+	onChange: ( value: Pick< ProductCategory, 'id' | 'name' >[] ) => void;
+};
+
+function getSelectedWithParents(
+	selected: Pick< ProductCategory, 'id' | 'name' >[] = [],
+	item: ProductCategory,
+	treeKeyValues: Record< number, CategoryTreeItem >
+): Pick< ProductCategory, 'id' | 'name' >[] {
+	selected.push( { id: item.id, name: item.name } );
+
+	const parentId = item.parent
+		? item.parent
+		: treeKeyValues[ item.id ].parentID;
+	if (
+		parentId > 0 &&
+		treeKeyValues[ parentId ] &&
+		! selected.find(
+			( selectedCategory ) => selectedCategory.id === parentId
+		)
+	) {
+		getSelectedWithParents(
+			selected,
+			treeKeyValues[ parentId ].data,
+			treeKeyValues
+		);
+	}
+
+	return selected;
+}
+
+export const CategoryField: React.FC< CategoryFieldProps > = ( {
+	label,
+	placeholder,
+	value = [],
+	onChange,
+} ) => {
+	const {
+		isSearching,
+		categoriesSelectList,
+		categoryTreeKeyValues,
+		searchCategories,
+		getFilteredItems,
+	} = useCategorySearch();
+	const [ showCreateNewModal, setShowCreateNewModal ] = useState( false );
+	const [ searchValue, setSearchValue ] = useState( '' );
+
+	const onInputChange = ( searchString?: string ) => {
+		setSearchValue( searchString || '' );
+		searchCategories( searchString || '' );
+	};
+
+	const searchDelayed = useMemo(
+		() => debounce( onInputChange, 150 ),
+		[ onInputChange ]
+	);
+
+	const onSelect = ( item: ProductCategory, selected: boolean ) => {
+		if ( selected ) {
+			onChange(
+				getSelectedWithParents(
+					[ ...value ],
+					item,
+					categoryTreeKeyValues
+				)
+			);
+		} else {
+			onChange( value.filter( ( i ) => i.id !== item.id ) );
+		}
+	};
+
+	const selectedIds = value.map( ( item ) => item.id );
+	let selectControlItems = categoriesSelectList;
+	// Add the add new button if search value does not exist.
+	if (
```

Comment:
This logic might be slightly dirty, I wasn't sure what the other easier method would be, but this only handles adding or removing the `Create "new category"` item from the list.